### PR TITLE
Restructure PhaseField

### PR DESCRIPTION
**Pull request by endJunction**
Same as #1877 and #1855 and https://github.com/ufz/ogs/pull/1937  to keep the processes similar.

### CODE REVIEW

File: ProcessLib/PhaseField/PhaseFieldFEM.h [hunk 1]
```diff
@@ -419,20 +338,22 @@ class PhaseFieldLocalAssembler : public PhaseFieldLocalAssemblerInterface
...
+                                                              phasefield_index)
             .noalias() += Kdd + Ddd / dt;
 
         // phasefield equation, displacement part.
-        local_Jac.template block<phasefield_size, displacement_size>(
-                     phasefield_index, displacement_index)
+        local_Jac
+            .template block<phasefield_size, displacement_size>(
+                phasefield_index, displacement_index)
             .noalias() += Kdu;
```
Comment (endJunction) [inline ProcessLib/PhaseField/PhaseFieldFEM.h]: @nagelt Above in line 317, in the if-condition (history_prev < strain_energy), the value of `Kdu` is set depending on the last integration point values for the history_prev and strain_energy. Is it correct? Seems to me like the results would depend on the ordering of the integration points.

File: ProcessLib/PhaseField/LocalAssemblerInterface.h [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+namespace ProcessLib
+{
+namespace PhaseField
+{
+
+struct PhaseFieldLocalAssemblerInterface
+    : public ProcessLib::LocalAssemblerInterface,
+      public NumLib::ExtrapolatableElement
+{
+    virtual std::vector<double> const& getIntPtSigmaXX(
```
Comment (chleh) [inline ProcessLib/PhaseField/LocalAssemblerInterface.h]: I suggest to also switch to vectorial output in this PR.

Comment (endJunction) [reply ProcessLib/PhaseField/LocalAssemblerInterface.h]: There will be a separate PR switching mechanics processes to vectorial output.

Comment (chleh) [reply ProcessLib/PhaseField/LocalAssemblerInterface.h]: That's also OK.

File: ProcessLib/PhaseField/PhaseFieldFEM.h (hunk 1 above)
Comment (xingyuanmiao) [reply ProcessLib/PhaseField/PhaseFieldFEM.h]: both the history_variable_prev and strain_energy_tensile are integration point data. The if-condition solves Kdu on each integration point.

Comment (nagelt) [reply ProcessLib/PhaseField/PhaseFieldFEM.h]: @xingyuanmiao not quite ... the if-condition determines whether or not the term is relevant in each integration point .. that's ok ... but the assigment you make then takes the entire K_ud transpose which has been assembled for all previous integration points (+=) ... so after your if you add more than you need ... and this leads to the randomness Dima describes ...

Comment (endJunction) [reply ProcessLib/PhaseField/PhaseFieldFEM.h]: I made the corresponding change in this PR. See commit `[PL] PhaseField: Fix computation of Kdu.`

**Comment by endJunction** [comment]
@bilke In Jenkins one of the builds could not connect to a docker; appveyor shows same behaviour with the conan's remote ogs. Could you please have a look?

**Comment by bilke** [comment]
@endJunction Conan Boost issue should be resolved: Conan cache directory was not writable by Jenkins on the new `azure-docker1`. I restarted the job..

**Pull request merged**